### ISSUE TITLE
Scan Schedule stuck in Disabled state

### ISSUE THREAD

**Issue by rstuberg-ib** [goharbor/harbor#14251]
If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
A clear and concise description of what you expected to happen and what's the actual behavior. If applicable, add screenshots to help explain your problem.
The scan api is stuck in a state where it is constantly trying to scan nothing. I don't remember if I started a schedule scan before or after I had images in there.

Started when we first deployed 2.0.0 upgraded to 2.1.3 and then the new 2.2.0-rc1/2. Still have the same problem.

![image](https://user-images.githubusercontent.com/30842306/108253944-3d30f980-710f-11eb-813b-2a5f136b1e20.png)

Button is disabled in UI. 

**Steps to reproduce the problem:**
Please provide the steps to reproduce this problem.

**Versions:**
Please specify the versions of following systems.
2.0->2.2-rc
Version v2.2.0-ef68ef30 is what I am trying to fix now.

- harbor version: [2.x.x]
- docker engine version: [19.03.13]

**Additional context:**
Curling the harbor scan api: 
```{
  "completed": 0,
  "metrics": {
    "Error": 0,
    "Pending": 0,
    "Running": 0,
    "Stopped": 0,
    "Success": 0
  },
  "ongoing": true,
  "total": 0,
  "trigger": "Schedule"
}```

- **Harbor config files:** You can get them by packaging `harbor.yml` and files in the same directory, including subdirectory.
I have Trivy and another scanner as a config. This started with the default config and Trivy having the problem. 

- **Log files:** You can get them by package the `/var/log/harbor/` .
I see no relevant logs from job service or core.

**Comment by rstuberg-ib** [comment]
I am also seeing problems with the gc. When I run gc it finishes quick => click logs in ui and get an empty response only logs I can see:

```
[ERROR] [/jobservice/api/handler.go:288]: Serve http request 'GET /api/v1/jobs/187238bd8021635391b84b14/log' error: 500 {"code":10009,"message":"failed to get the job log","details":"error for getting log of job 187238bd8021635391b84b14: {\"code\":10010,\"message\":\"object is not found\",\"details\":\"187238bd8021635391b84b14\"}"}
```

**Comment by rstuberg-ib** [comment]
The DB is littered with these records:
                                                                 | 2021-02-16 16:00:01.0
22792 |                            |        0 | 2021-02-16 16:00:01.022792
 132 | SCAN_ALL           |         0 | Running |                                    | SC
HEDULE | {}

**Comment by heww** [comment]
Hi, @rstuberg-ib could you share the completed response of the scan all metric APIs?

Two scan all metric APIs are called on this page, `/api/v2.0/scans/schedule/metrics` and `/api/v2.0/scans/all/metrics`

**Comment by rstuberg-ib** [comment]
api/v2.0/scans/all/metrics
```
{
  "completed": 0,
  "metrics": {
    "Error": 0,
    "Pending": 0,
    "Running": 0,
    "Stopped": 0,
    "Success": 0
  },
  "ongoing": true,
  "total": 0,
  "trigger": "Manual"
}
```
api/v2.0/scans/schedule/metrics
```
{
  "completed": 0,
  "metrics": {
    "Error": 0,
    "Pending": 0,
    "Running": 0,
    "Stopped": 0,
    "Success": 0
  },
  "ongoing": false,
  "total": 0,
  "trigger": "Schedule"
}
```

**Comment by rstuberg-ib** [comment]
I don't know if this is a common symptom or not too but when going to GC it finishes instantly and the logs are empty and doesn't return content.

**Comment by heww** [comment]
Do these problems happen in harbor 2.2-rc2? And how many artifacts in your harbor system?

**Comment by heww** [comment]
> The DB is littered with these records:
> ```
> | 2021-02-16 16:00:01.0
> 22792 | | 0 | 2021-02-16 16:00:01.022792
> 132 | SCAN_ALL | 0 | Running | | SC
> HEDULE | {}

According to the source code of the scan all, the extra_attrs column of the execution for the `SCAN_ALL` will never be empty, except the updating of the extra_attrs failed.

```go
	extraAttrs := map[string]interface{}{"summary": summary}
	if err := bc.execMgr.UpdateExtraAttrs(ctx, executionID, extraAttrs); err != nil {
		log.Errorf("failed to set the summary info for the scan all execution, error: %v", err)
		return err
	}
```

Could you search to the log of the core to check whether `failed to set the summary info for the scan all execution` be found in the log?

**Comment by rstuberg-ib** [comment]
I did a fresh redeploy and still am running into the issue:

```
select * from execution where vendor_type='SCAN_ALL';
 id | vendor_type | vendor_id | status  | status_message | trigger | extra_attr
s |         start_time         | end_time | revision |        update_time

----+-------------+-----------+---------+----------------+---------+-----------
--+----------------------------+----------+----------+-------------------------
---
  2 | SCAN_ALL    |         0 | Error   |                | MANUAL  | {}
  | 2021-02-18 21:20:52.606855 |          |        0 | 2021-02-18 21:20:52.6068
55
 75 | SCAN_ALL    |         0 | Running |                | MANUAL  | {}
  | 2021-02-18 22:38:25.128039 |          |        0 | 2021-02-18 22:38:25.1280
39
(2 rows)
```

I manually updated the table to Error out the other scan_all job to see if it would do anything. It was sitting in Running state for 5+ hours not adding anything to pending etc.

All of the images are replicated in (I don't know if that would matter)

I don't see that log in the core harbor container.

**Comment by rstuberg-ib** [comment]
```
select * from task where vendor_type = 'SCAN_ALL'
registry-> ;
 id | execution_id | job_id | status | status_code | status_revision | status_m
essage | run_count | extra_attrs | creation_time | start_time | update_time | e
nd_time | vendor_type
----+--------------+--------+--------+-------------+-----------------+---------
-------+-----------+-------------+---------------+------------+-------------+--
--------+-------------
(0 rows)
```

```
select * from schedule;
 id | creation_time | update_time | vendor_type | vendor_id | cron | callback_f
unc_name | callback_func_param | cron_type | extra_attrs
----+---------------+-------------+-------------+-----------+------+-----------
---------+---------------------+-----------+-------------
(0 rows)
```

**Comment by heww** [comment]
The problem looks like the scan all execution was created but no scan task submitted.

Could you paste the log of the core component when executing the scan all?

**Comment by rstuberg-ib** [comment]
Can only see this in DEBUG log level
`
[harbor-core] panic: runtime error: invalid memory address or nil pointer dereference
[harbor-core] [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbd7258]
[harbor-core]
[harbor-core] goroutine 1424 [running]:
[harbor-core] github.com/goharbor/harbor/src/controller/artifact.Iterator.func1(0xc000d2d4a0, 0x0, 0x32, 0x26e5680, 0xc000cbd0b0, 0x0)
[harbor-core] 	/harbor/src/controller/artifact/helper.go:50 +0xf8
[harbor-core] created by github.com/goharbor/harbor/src/controller/artifact.Iterator
[harbor-core] 	/harbor/src/controller/artifact/helper.go:28 +0x8f
`

**Comment by heww** [comment]
Hi @rstuberg-ib could you help to verify that this issue was fixed in [v2.2.0-rc4 ](https://github.com/goharbor/harbor/releases/tag/v2.2.0-rc4)? Thanks!

**Comment by rstuberg-ib** [comment]
@heww this is working now.  Thanks for the quick turnaround!
